bin/metadata_validation.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+import json
+
+import synapseclient
+from synapseclient import Entity
+
+
+def get_synaspe_file(syn_id):
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** You can validate the `synid` variable against a regex like `syn[0-9]+`.

Reply:
Neither of those desired behaviors is happening in the current version. If I set `cache` to 'deep', then SYNAPSE_GET is prevented from running again when `-resume` is used. New versions do not trigger re-runs, even with `cache 'deep'`.

Reply:
What do you suggest should happen if a `syn_id` does not conform to the regex pattern?

Reply:
In the latest commit, I added an `assert` statement to validate the Synapse ID in the `get_synapse_file` function. Let me know if you think a more robust solution is needed (for example, actually handling cases with invalid synapse IDs without just throwing an error and stopping the workflow)